### PR TITLE
Remove default Java toolchain downloaders (AdoptOpenJdk/Adoptium)

### PR DESCRIPTION
**Pull request by jbartok**

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
Would be good to officially release the Foojay Disco plugin to not have to use it from the DEV portal and to make the repo open source.

**Comment by jbartok** [comment]
@bot-gradle test QFL

File: subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSoakTest.groovy [hunk 1]
```diff
@@ -21,14 +21,38 @@ import org.gradle.integtests.fixtures.AbstractIntegrationSpec
...
+            
+            toolchainManagement {
+                jvm {
+                    javaRepositories {
+                        repository('disco') {
+                            resolverClass = org.gradle.disco.DiscoToolchainResolver
+                        }
+                    }
+                }
+            }
```
Comment (octylFractal) [inline subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSoakTest.groovy:45]: This makes me concerned about this change -- this is _incredibly_ verbose for what used to be a one-line feature, which is likely to put people off of using it. I know I would be less likely to use toolchains if I had to copy around 12 lines to every single project I made. Is there any way that we could have a smaller way to enable this?

Comment (jbartok) [reply subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSoakTest.groovy:45]: You are right, we are addressing this with a convenience plugin, published together with the basic foojay plugin, which, when applied, will automatically add the toolchain management part.

**Comment by jbartok** [comment]
@bot-gradle test QFL

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

File: subprojects/docs/src/snippets/java/toolchain-management/groovy/settings.gradle [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 
 // tag::toolchain-management[]
 toolchainManagement {
     jvm { // <1>
         javaRepositories {
-            repository('azul') { // <2>
-                resolverClass = AzulResolver
+            repository('disco') { // <2>
+                resolverClass = org.gradle.toolchains.foojay.FoojayToolchainResolver
                 credentials {
```
Comment (ljacomet) [inline subprojects/docs/src/snippets/java/toolchain-management/groovy/settings.gradle:16]: We should move that to the `adoptium` element, as it makes no sense for our foojay plugin

File: subprojects/enterprise-operations/src/main/java/org/gradle/jvm/toolchain/internal/operations/JavaToolchainUsageProgressDetails.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ interface JavaToolchain {
         String getJvmVersion();
 
         /**
-         * Returns Java VM vendor such as {@code Eclipse Adoptium}.
+         * Returns Java VM vendor such as {@code Oracle}.
```
Comment (ljacomet) [inline subprojects/enterprise-operations/src/main/java/org/gradle/jvm/toolchain/internal/operations/JavaToolchainUsageProgressDetails.java:75]: Why that change?

**Review by ljacomet — COMMENTED** [review]
The removal makes sense, but there are changes I do not understand, see below.

File: build-logic-commons/gradle-plugin/src/main/kotlin/common.kt [hunk 4]
```diff
@@ -15,11 +15,9 @@
  */
 import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.jvm.toolchain.JavaLanguageVersion
-import org.gradle.jvm.toolchain.JvmVendorSpec
 
 fun JavaPluginExtension.configureJavaToolChain() {
     toolchain {
         languageVersion.set(JavaLanguageVersion.of(11))
-        vendor.set(JvmVendorSpec.ADOPTIUM)
```
[the remark this answers is not in the document]
Comment (jbartok) [reply build-logic-commons/gradle-plugin/src/main/kotlin/common.kt:23]: I don't have a better reason than too eager to remove everything that mentions Adoptium and I could revert it, but what do we gain by specifying a vendor there?

File: subprojects/docs/src/snippets/java/toolchain-management/groovy/settings.gradle (hunk 2 above)
Comment (jbartok) [reply subprojects/docs/src/snippets/java/toolchain-management/groovy/settings.gradle:16]: You mean the authentication bits, right? Moving it.

File: build-logic-commons/gradle-plugin/src/main/kotlin/common.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ljacomet) [reply build-logic-commons/gradle-plugin/src/main/kotlin/common.kt:23]: As discussed, let's revert this change and the one to `gradlebuild.unittest-and-compile.gradle.kts`

**Review by ljacomet — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle test this

**Pull request merged**